accel-pppd/ctrl/sstp/sstp.c

```diff
@@ -2584,21 +2584,20 @@ static void ssl_load_config(struct sstp_serv_t *serv, const char *servername)
 #ifdef OPENSSL_NO_DH
 			log_warn("sstp: %s warning: %s is not suported\n", "ssl-protocol", "DH");
 #else
-			DH *dh;
+			EVP_PKEY *dh;
 
 			if (BIO_read_filename(in, opt) <= 0) {
 				log_error("sstp: %s error: %s\n", "ssl-dhparam", ERR_error_string(ERR_get_error(), NULL));
 				goto error;
 			}
 
-			dh = PEM_read_bio_DHparams(in, NULL, NULL, NULL);
+			dh = PEM_read_bio_Parameters(in, NULL);
 			if (dh == NULL) {
 				log_error("sstp: %s error: %s\n", "ssl-dhparam", ERR_error_string(ERR_get_error(), NULL));
 				goto error;
 			}
 
-			SSL_CTX_set_tmp_dh(ssl_ctx, dh);
-			DH_free(dh);
+			SSL_CTX_set0_tmp_dh_pkey(ssl_ctx, dh);
```

Comment:
Please consider checking the return value of SSL_CTX_set0_tmp_dh_pkey() to correctly to avoid memory leaks and surface configuration errors early.
```suggestion
	if (SSL_CTX_set0_tmp_dh_pkey(ssl_ctx, dhpkey) != 1) {
		log_error("sstp: %s error: %s\n",
		          "ssl-dhparam",
		          ERR_error_string(ERR_get_error(), NULL));
		EVP_PKEY_free(dhpkey);
		goto error;
	}
```